### PR TITLE
allow linking, posix and spaces conversions in datalib imports

### LINKED ISSUE #1529: Feature request - symlink files when uploading data to data library using the "from user directory" option
Hi, 

When using the "user_library_import_dir" option user get the possibility to upload files into data libraries. However, it's currently not possible to symlink those files (instead of copying) when selecting a single file. If the user wants to upload an entire folder this option is available.

Is it possible to add the "symlink" checkbox to the upload interface?

Thanks!
Matthias

### LINKED ISSUE #2977: enhance libraries UI with newline and tab conversion options
Both the upload intrface and the old data libraries UI have options to `convert tabs to spaces` and to `convert line endings to POSIX`. The new libraries UI would benefit from these options too.

The conversion functionality is already present in the upload handling so we just need checkboxes in the UI to propagate the params.

The following PR implemented the POSIX conversion for the old UI, the PR to close this issue would be very similar to that.

xref: https://github.com/galaxyproject/galaxy/pull/2819

### PR DESCRIPTION
**Pull request by martenson**
closes #2977

closes https://github.com/galaxyproject/galaxy/issues/1529

### CODE REVIEW

File: client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js [hunk 1]
```diff
@@ -844,11 +857,13 @@ var FolderToolbarView = Backbone.View.extend({
...
 
   /**
-   * Take the array of paths and createa request for each of them
-   * calling them in chain. Update the progress bar in between each.
+   * Take the array of paths and create a request for each of them
+   * calling them in series. Update the progress bar in between each.
    * @param  {array} paths                    paths relative to Galaxy root folder
    * @param  {boolean} preserve_dirs          indicates whether to preserve folder structure
    * @param  {boolean} link_data              copy files to Galaxy or link instead
+   * @param  {boolean} to_posix_lines              copy files to Galaxy or link instead
```
Comment (dannon) [inline client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js]: Comment should be updated here probably (copied from above when editing, looks like)

Comment (martenson) [reply client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js]: resolved, thanks!

**Comment by martenson** [comment]
So, make client has no output on comment changes...but shouldn't it build a new map?

**Comment by dannon** [comment]
@martenson I read through https://sourcemaps.info/spec.html and it seems to me like the mapping offsets *should* stay the same when only the comments change like that, so this looks good to me.

**Comment by martenson** [comment]
@dannon kudos to them, good design

**Pull request merged**